src/ptunnel.c

```diff
@@ -558,10 +557,23 @@ void* pt_proxy(void *args) {
 			if (cur->state == kProxy_start) {
 				pt_log(kLog_verbose, "Sending proxy request.\n");
 				cur->last_ack = time_as_double();
-				queue_packet(fwd_sock, cur->pkt_type, 0, 0, cur->id_no, cur->id_no,
+				uint16_t *extended_options = 0;
+				size_t extended_options_size = 0;
+				if (opts.window_size || opts.ack_interval || opts.resend_interval || opts.payload_size) {
```

Comment:
Checking only for *one* !NULL option but using all non-checked opts for rvalues?

Reply:
Options are initialized to zero in options.c. Extended options are sent by the client if any value is non-zero. Values are checked by the server when they are received in pkt.c. Values are checked before they are used in pdesc.c where zero indicates defaults.